src/linux/Packaging.Linux/install-from-source.sh

```diff
@@ -0,0 +1,176 @@
+
+    /usr/local/bin
+
+Git Credential Manager is licensed under the MIT License: https://aka.ms/gcm/license"
+
+    while true; do
+        read -p "Do you want to continue? [Y/n] " yn
+        case $yn in
+            [Yy]*|"")
+                break
+            ;;
+            [Nn]*)
+                exit
+            ;;
+            *) 
+                echo "Please answer yes or no."
+            ;;
+        esac
+    done
+fi
+
+install_shared_packages() {
+    pkg_manager=$1
+    install_verb=$2
+
+    local shared_packages="git curl"
+    for package in $shared_packages; do
+        if [ $pkg_manager = apk ]; then
+            $sudo_cmd $pkg_manager $install_verb $package
+        else
+            $sudo_cmd $pkg_manager $install_verb $package -y
+        fi
+    done
+}
+
+ensure_dotnet_installed() {
+    # get initial pieces of installed sdk version(s)
+    sdks=$(dotnet --list-sdks | cut -c 1-3)
+
+    # if we have a supported version installed, no need to
+    # download and run the script
+    supported_dotnet_versions="6.0 5.0"
+    for v in $supported_dotnet_versions; do
+        if [ $(echo $sdks | grep "$v") ]; then
+            return
+        fi
+    done
+
+    curl -LO https://dot.net/v1/dotnet-install.sh
+    chmod +x ./dotnet-install.sh
+    bash -c "./dotnet-install.sh"
+
+    # since we have to run the dotnet install script with bash, dotnet isn't added
+    # to the process PATH, so we manually add it here
+    cd ~
+    export DOTNET_ROOT=$(pwd)/.dotnet
+    add_to_PATH $DOTNET_ROOT
+}
+
+add_to_PATH () {
+  for directory; do
+    if [ ! -d "$directory" ]; then
+        continue; # skip nonexistent directory
+    fi
+    case ":$PATH:" in
+        *":$directory:"*)
+            break
+        ;;
+        *)
+            export PATH=$PATH:$directory
+        ;;
+    esac
+  done
+}
+
+sudo_cmd=
+
+# if the user isn't root, we need to use `sudo` for certain commands
+# (e.g. installing packages)
+if [ -z "$sudo_cmd" ]; then
+    if [ `id -u` != 0 ]; then
+        sudo_cmd=sudo
+    fi
+fi
+
+eval "$(sed -n 's/^ID=/distribution=/p' /etc/os-release)"
+eval "$(sed -n 's/^VERSION_ID=/version=/p' /etc/os-release | tr -d '"')"
+case "$distribution" in
+    debian | ubuntu)
+        $sudo_cmd apt update
+        install_shared_packages apt install
+
+        # add dotnet package repository/signing key
+        $sudo_cmd apt update && $sudo_cmd apt install wget -y
+        curl -LO https://packages.microsoft.com/config/"$distribution"/"$version"/packages-microsoft-prod.deb
+        $sudo_cmd dpkg -i packages-microsoft-prod.deb
+        rm packages-microsoft-prod.deb
+
+        # proactively install tzdata to prevent prompts
+        export DEBIAN_FRONTEND=noninteractive
+        $sudo_cmd apt install -y --no-install-recommends tzdata
+
+        # install dotnet packages and dependencies
+        $sudo_cmd apt update
+        $sudo_cmd apt install dotnet-sdk-5.0 dpkg-dev apt-transport-https -y
```

Comment:
```suggestion
        # install dotnet packages and dependencies
        $sudo_cmd apt update
        $sudo_cmd apt install apt-transport-https -y
        $sudo_cmd apt update
        $sudo_cmd apt install dotnet-sdk-5.0 dpkg-dev -y
```

(Suggested change based on [official Microsoft docs](https://docs.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#2110-))